instrumentation/servlet/servlet-3.0/javaagent/src/test/groovy/AbstractServlet3MappingTest.groovy

```diff
@@ -48,9 +49,12 @@ abstract class AbstractServlet3MappingTest<SERVER, CONTEXT> extends AgentInstrum
         span(0) {
           name getContextPath() + spanName
           kind SpanKind.SERVER
-          if (!success) {
+          if (!success && response.status().code() >= 500) {
             status ERROR
           }
+          else if (!success && response.status().code() < 500) {
+            status UNSET
+          }
```

Comment:
Nit: no need to check for that explicitly; `SpanAssert` will verify that the status is `UNSET` in case you haven't done that in your test code

Reply:
@breedx-splk is this another vote for converting groovy tests to java? 😂